packages/@ember/-internals/container/lib/registry.ts

```diff
@@ -417,7 +417,7 @@ export default class Registry implements IRegistry {
 
   getOptions(fullName: string): TypeOptions | undefined {
     let normalizedName = this.normalize(fullName);
-    let options = this._options[normalizedName];
+    let options = normalizedName in this._options ? this._options[normalizedName] : undefined;
```

Comment:
Hmm. This shouldn't be complaining about the access. 🤔 (See [playground][play].) Since it's a string index, and given the type you wrote, it *should* correctly resolve it in either the `Record` or the index type form. What error was TS giving you here?

[play]: https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCoE8AOEDymzAD2IAzsgN4BQyymhYcYhA-AFzIlhSgDmA3JQC+lSqEixEKHGAAW0CtWQB3GYwCqmNsgDkICI0LaBwyggA2cEmQBihQgpqZuAN0YoA+oXxFS7AEoQCIRQACYAPJzcIDwANGhYuN7EZAA+yACuICEQMKAQIQB8yAC8yDgARgBWgWAAdAhQ+pAAFCDpZmYAlAKOLm7InrLQ7OTIANoA1hDo7JG8ALrs0nJQyGmZ2bl6IciCJWVVNfWNbq3tXQKKPBBgeATJzTDnAHJwALYQs1y8newY2HcfKkMlkcnkdlQaDQgqQwMgvPdSPtZMASLVPElSGMnh1Xh95j0oY0wOkoCB4ZiSISTDRrrchlBHi93p8ON9or8ygy1iDNuCHESbqTySi0YMVtjmfjqUJKEA

Reply:
Not sure what was happening before but it seems ok now.